bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/AbstractComponent.java

```diff
@@ -76,15 +82,30 @@ public abstract class AbstractComponent<C extends AbstractChannelConfiguration>
     protected final C channelConfiguration;
 
     protected boolean configSeen;
+    protected final boolean newStyleChannels;
+    protected final boolean singleChannelComponent;
+    protected final String groupId;
+    protected final String uniqueId;
+
+    public AbstractComponent(ComponentFactory.ComponentConfiguration componentConfiguration, Class<C> clazz,
+            boolean newStyleChannels) {
+        this(componentConfiguration, clazz, newStyleChannels, false);
+    }
 
     /**
      * Creates component based on generic configuration and component configuration type.
      *
      * @param componentConfiguration generic componentConfiguration with not parsed JSON config
      * @param clazz target configuration type
+     * @param newStyleChannels if new style channels should be used
+     * @param singleChannelComponent if this component only ever has one channel, so should never be in a group
+     *            (only if newStyleChannels is true)
      */
-    public AbstractComponent(ComponentFactory.ComponentConfiguration componentConfiguration, Class<C> clazz) {
+    public AbstractComponent(ComponentFactory.ComponentConfiguration componentConfiguration, Class<C> clazz,
+            boolean newStyleChannels, boolean singleChannelComponent) {
```

Comment:
Is this possible to move newStyleChannels into haID?

Reply:
before I spend time making changes here, I'd like feedback on how we want to address backwards compatibility:
 * as is - newly discovered things get the shorter channels, already-existing things don't. file-based things don't (which... I don't even know if file-based Home Assistant things really work, so might not matter)
 * an explicit configuration on the thing to opt in (means can be used for file-based things, or if you have to re-create your things for some reason) - but means it's an additional step, so most people won't ever know about it. and not really useful for re-creating things, since the thing uid will always be the shorter version, even if you choose the longer channel ids.
 * an explicit configuration on the thing to opt out (means can be used for file-based things, or if you have to re-create your things for some reason) - but means it will break all existing things causing issues for existing users anyway.  and not really useful for re-creating things, since the thing uid will always be the shorter version, even if you choose the longer channel ids.
 * just pull off the band aid and call a breaking change - all existing things get the newer shorter ids, and no way to go back. definitely simplifies the code to not need to plumb this through, or to have to have two branches

Reply:
Yes, that is the best way to announce that this change is coming.